concourse/steps/scan_sources.py

```diff
@@ -63,98 +66,156 @@ def scan_component_with_whitesource(
 
     # create whitesource client
     ci.util.info('creating whitesource client')
-    client = whitesource.util.create_whitesource_client(whitesource_cfg_name=whitesource_cfg_name)
+    ws_client = whitesource.util.create_whitesource_client(
+        whitesource_cfg_name=whitesource_cfg_name,
+    )
 
     # parse component_descriptor
     ci.util.info('parsing component descriptor')
     component_descriptor = cm.ComponentDescriptor.from_dict(
         ci.util.parse_yaml_file(component_descriptor_path)
     )
 
-    # for component in []: 'components matching component_descriptor'
-    for component in product.v2.components(component_descriptor_v2=component_descriptor):
+    product_name = component_descriptor.component.name
+
+    components = product.v2.components(component_descriptor_v2=component_descriptor)
+
+    # get scan artifacts with configured label
+    scan_artifacts_gen = whitesource.component._get_scan_artifacts_from_components(components)
+
+    scan_artifacts = tuple(scan_artifacts_gen)
+    ci.util.info(f'will scan {len(scan_artifacts)} artifact')
+
+    for scan_artifact in scan_artifacts:
+        scan_artifact_with_ws(
+            extra_whitesource_config=extra_whitesource_config,
+            product_token=product_token,
+            requester_mail=requester_mail,
+            scan_artifact=scan_artifact,
+            ws_client=ws_client,
+        )
+
+    notify_users(
+        notification_recipients=notification_recipients,
+        cve_threshold=cve_threshold,
+        ws_client=ws_client,
+        product_name=product_name,
+        product_token=product_token,
+    )
+
+
+def scan_artifact_with_ws(
```

Comment:
hmm... this is a lot of code in our template. we should consider moving it into the `whitesource`-pkg, maybe (keep the interface between template and our libs small..)

Reply:
moved the code to the util package